lib/Service/RoomService.php

```diff
@@ -524,4 +542,43 @@ public function verifyPassword(Room $room, string $password): array {
 			'url' => '',
 		];
 	}
+
+	public function setExpireInterval(Room $room, Participant $participant, int $seconds): void {
+		$event = new ChangeExpireDateEvent($room, $seconds);
+		$this->dispatcher->dispatch(Room::EVENT_BEFORE_SET_EXPIRE_DATE, $event);
+
+		$update = $this->db->getQueryBuilder();
+		$update->update('talk_rooms')
+			->set('message_expire', $update->createNamedParameter($seconds, IQueryBuilder::PARAM_INT))
+			->where($update->expr()->eq('id', $update->createNamedParameter($room->getId(), IQueryBuilder::PARAM_INT)));
+		$update->executeStatement();
+		$room->setExpireInterval($seconds);
+		if ($seconds > 0) {
+			$this->jobList->add(ApplyExpireDate::class, ['room_id' => $room->getId()]);
+			$this->expireDateSystemMessage($room, $participant, $seconds, 'message_expire_enabled');
+		} else {
+			$this->jobList->remove(ApplyExpireDate::class, ['room_id' => $room->getId()]);
```

Comment:
There could still be messages which need to be expired.
So we need to keep the job?

Wasn't that also the reason why we wanted to move to a single background job instead of "per conversation"?

Reply:
Probably we don't need to maintain this. We have here two scenarios:

Expire job by rom:
- We only will have a job if we have a room with expire message time.

Global expire message:
- A query running every 5 minutes (or other interval defined by cron)

Thinking about performance, I believe that if we have a job running all times that the cron is executed, won't be a problem.

What way you preffer?

But is working now, maybe we can merge this PR for the developers of clients go ahead and modify this in a second step in other PR to don't make this PR more big.

Reply:
> We only will have a job if we have a room with expire message time.

That is what I'm trying to say. It is not true. Expiration could be turned off now, but messages that where posted before still need to expire like the setting was back then. so you can not immediatelly remove the background job.

Reply:
> But is working now, maybe we can merge this PR for the developers of clients go ahead and modify this in a second step in other PR to don't make this PR more big.

yeah